src/snowflake/snowpark/_internal/utils.py

```diff
@@ -1924,6 +1925,141 @@ def remove_comments(sql_query: str, uuids: List[str]) -> str:
     )
 
 
+class TimeTravelConfig(NamedTuple):
+    """Configuration for time travel operations."""
+
+    mode: str  # 'at' or 'before'
+    statement: Optional[str] = None
+    offset: Optional[int] = None
+    timestamp: Optional[str] = None
+    timezone: str = "NTZ"
+    stream: Optional[str] = None
+
+
+def validate_and_normalize_time_travel_params(
+    time_travel_mode: Optional[str] = None,
+    statement: Optional[str] = None,
+    offset: Optional[int] = None,
+    timestamp: Optional[Union[str, datetime.datetime]] = None,
+    timezone: Optional[str] = "NTZ",
```

Comment:
I'm wondering if we can also allow users to use `types.TimestampTimeZone` to specify timezone